util/azure/utils.go

```diff
@@ -0,0 +1,341 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	"go.kubeguard.dev/guard/auth/providers/azure/graph"
+	"go.kubeguard.dev/guard/util/httpclient"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pkg/errors"
+	v "gomodules.xyz/x/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+const (
+	ManagedClusters             = "Microsoft.ContainerService/managedClusters"
+	Fleets                      = "Microsoft.ContainerService/fleets"
+	ConnectedClusters           = "Microsoft.Kubernetes/connectedClusters"
+	OperationsEndpointFormatARC = "%s/providers/Microsoft.Kubernetes/operations?api-version=2021-10-01"
+	OperationsEndpointFormatAKS = "%s/providers/Microsoft.ContainerService/operations?api-version=2018-10-31"
+	MSIEndpointForARC           = "http://127.0.0.1:8421/metadata/identity/oauth2/token?api-version=2018-02-01"
+)
+
+type TokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	ExpiresIn    string `json:"expires_in"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresOn    string `json:"expires_on"`
+	NotBefore    string `json:"not_before"`
+	Resource     string `json:"resource"`
+	TokenType    string `json:"token_type"`
+	ErrorMessage string `json:"error_message,omitempty"`
+}
+
+type Display struct {
+	Provider    string `json:"provider"`
+	Resource    string `json:"resource"`
+	Operation   string `json:"operation"`
+	Description string `json:"description"`
+}
+
+type Operation struct {
+	Name         string  `json:"name"`
+	Display      Display `json:"display"`
+	IsDataAction *bool   `json:"isDataAction,omitempty"`
+}
+
+type OperationList struct {
+	Value    []Operation `json:"value"`
+	NextLink string      `json:"nextLink"`
+}
+
+type Resource struct {
+	Id         string
+	Namespaced bool
+	Name       string
+	Group      string
+	Verb       string
+}
+
+type AuthorizationEntity struct {
+	Id string `json:"Id"`
+}
+
+type AuthorizationActionInfo struct {
+	AuthorizationEntity
+	IsDataAction bool `json:"IsDataAction"`
+}
+
+type DataAction struct {
+	ActionInfo           AuthorizationActionInfo
+	IsNamespacedResource bool
+}
+
+type ResourceAndVerbMap struct {
+	ResourceMap map[string]map[string]DataAction
+}
+
+type OperationsMap struct {
```

Comment:
I think https://go.dev/play/p/6VIr_D3wWmY will look much simpler?

Reply:
operationsMap.GroupMap[group].ResourceMap[resourceName][verb] = da will become much shorter to
operationsMap[group][resourceName][verb]